backend/s3/s3.go

```diff
@@ -2992,19 +2993,12 @@ func (f *Fs) NewObject(ctx context.Context, remote string) (fs.Object, error) {
 
 // Gets the bucket location
 func (f *Fs) getBucketLocation(ctx context.Context, bucket string) (string, error) {
-	req := s3.GetBucketLocationInput{
-		Bucket: &bucket,
-	}
-	var resp *s3.GetBucketLocationOutput
-	var err error
-	err = f.pacer.Call(func() (bool, error) {
-		resp, err = f.c.GetBucketLocation(&req)
-		return f.shouldRetry(ctx, err)
-	})
+	sess := session.Must(session.NewSession())
+	region, err := s3manager.GetBucketRegion(ctx, sess, bucket, "us-east-1")
```

Comment:
Reading the docs for s3manager.GetBucketRegion we should probably be forcing PathStyle = false, or perhaps better setting it to what it is set to in the config `f.opt.ForcePathStyle`

```
        region, err := s3manager.GetBucketRegion(ctx, sess, "bucketname", "us-west-2", func(r *request.Request) {
            r.S3ForcePathStyle = aws.Bool(f.opt.ForcePathStyle)
        })
```

Actually thinking on this a bit more, we should probably use use `f.ses` as the session as it will have all the right settings in already. It will do an authenticated request rather than unauthenticated, but I don't suppose that will be a problem.

Do you think the region hint should be the users current region in the config `f.opt.Region`? This will default to `us-east-1` if not set.

I don't really understand what the region hint is for!

Reply:
> Do you think the region hint should be the users current region in the config f.opt.Region?

Sure! I wasn't aware it was accessible in the context.

> I don't really understand what the region hint is for!

Me neither actually because using `curl` I can target `s3.amazonaws.com` directly and it works. I do not need to target a specific region's endpoint (ex: `s3.us-east-1.amazonaws.com`).

The SDK also clearly check for `regionHint` being empty:

https://github.com/aws/aws-sdk-go/blob/36e99e2d8b7f9efaa27e9665760788b3169e24bb/service/s3/s3manager/bucket_region.go#L65

But the comment also says:

```
// If the regionHint parameter is an
// empty string GetBucketRegion will fallback to the ConfigProvider's region
// config. If the regionHint is empty, and the ConfigProvider does not have a
// region value, an error will be returned..
```

If I use `f.ses` I might not need to pass the region here if I assume it is already configured with it.

Reply:
For `f.opt.ForcePathStyle` I'll do the change, I did not even see that flag already existed in rclone 🤦🏻

Reply:
PR updated with both changes for `f.ses` & `f.opt.ForcePathStyle` which is much cleaner 👍🏻